core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -181,6 +187,7 @@ public static RecordState forId(byte id) {
         // TODO: Just a placeholder for now.
         assert this.maxInFlightMessages > 0;
         assert this.maxDeliveryCount > 0;
+        this.fetchLock = new AtomicBoolean(true);
```

Comment:
Shouldn't it be `false` to start with?

Reply:
I had based the boolean value on the basis of availability of lock not acquiring process of lock, so that's why `true`. I agree it might be confusing so I will change the boolean values to opposite values